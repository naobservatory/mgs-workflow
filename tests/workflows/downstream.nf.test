nextflow_pipeline {

    name "Test workflow DOWNSTREAM"
    script "main.nf" // Running from main so we have access to published files
    config "tests/configs/downstream.config"
    tag "workflow"

    test("DOWNSTREAM workflow should run without errors") {
        tag "downstream"
        tag "main_downstream"
        then {
            assert workflow.success
        }
    }

    test("DOWNSTREAM workflow output should match snapshot") {
        tag "downstream_output"
        tag "main_downstream_output"
        tag "snapshot"
        then {
            assert workflow.success
            assert snapshot(
                path("${launchDir}/output/results_downstream/gs1_duplicate_reads.tsv.gz"),
                path("${launchDir}/output/results_downstream/gs1_duplicate_stats.tsv.gz"),
                path("${launchDir}/output/results_downstream/gs1_clade_counts.tsv.gz"),
                path("${launchDir}/output/results_downstream/gs1_validation_hits.tsv.gz"),
            ).match("downstream_output")
        }
    }

}
