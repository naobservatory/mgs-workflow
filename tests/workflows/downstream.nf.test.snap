{
    "downstream_output": {
        "content": [
            "gs1_duplicate_reads.tsv.gz:md5,1fec7b7960c4127f820426219d13466b",
            "gs1_duplicate_stats.tsv.gz:md5,48f655dffa173d1c8b23572ab2039e34",
            "gs1_clade_counts.tsv.gz:md5,e58c1489df2a1dc5df903c916c5c9ee7",
            "gs1_validation_hits.tsv.gz:md5,a88e96cd7c3f7215ac76a9737623b028"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-06T10:41:58.006912226"
    }
}